src/redis-cli.c

```diff
@@ -9099,7 +9099,9 @@ static void sendReadOnly(void) {
     if (read_reply == NULL){
         fprintf(stderr, "\nI/O error\n");
         exit(1);
-    } else if (read_reply->type == REDIS_REPLY_ERROR && strcmp(read_reply->str, "ERR This instance has cluster support disabled") != 0) {
+    } else if (read_reply->type == REDIS_REPLY_ERROR && 
+               strcmp(read_reply->str, "ERR This instance has cluster support disabled") != 0 &&
+               strcmp(read_reply->str, "ERR unknown command 'READONLY'") != 0) {
```

Comment:
older versions of redis didn't have the exact `ERR unknown command 'READONLY'` reply.
i think the quotes were missing, and the command name could have been lower case.
i suggest to match just the prefix `ERR unknown command`.